app/scripts/metamask-controller.js

```diff
@@ -1012,9 +1012,17 @@ export default class MetamaskController extends EventEmitter {
     });
     this.smartTransactionsController = new SmartTransactionsController(
       {
-        onNetworkStateChange: this.networkController.store.subscribe.bind(
-          this.networkController.store,
-        ),
+        onNetworkStateChange: (cb) => {
+          this.networkController.store.subscribe((networkState) => {
+            const modifiedNetworkState = {
+              ...networkState,
+              providerConfig: {
```

Comment:
the `smart-transaction-controller` types the networkState using a type from `@metamask/network-controller` (from `@metamask/core-monorepo`) that was recently updated to expect property `providerConfig` in place of `provider`: https://github.com/MetaMask/core/blob/main/packages/network-controller/CHANGELOG.md?plain=1#L11